src/transform/benches/mzbench_opt/cli.py

```diff
@@ -0,0 +1,189 @@
+
+    db_user: str = typer.Option("materialize", help="DB connection user.")
+
+
+@app.command()
+def init(
+    scenario: Scenario = Arg.scenario,
+    db_port: int = Opt.db_port,
+    db_host: str = Opt.db_host,
+    db_user: str = Opt.db_user,
+) -> None:
+    """Initialize the DB under test for the given `scenario`."""
+
+    info(f'Initializing "{scenario}" as the DB under test')
+
+    try:
+        db = sql.Database(port=db_port, host=db_host, user=db_user)
+
+        db.drop_database(scenario)
+        db.create_database(scenario)
+
+        db.set_database(scenario)
+        db.execute_all(statements=sql.parse_from_file(sql.schema_path(scenario)))
+    except Exception as e:
+        err(f"DB initialization failed: {e}")
+        raise typer.Exit()
+
+
+@app.command()
+def run(
+    scenario: Scenario = Arg.scenario,
+    samples: int = Opt.samples,
+    repository: Path = Opt.repository,
+    print_results: bool = Opt.print_results,
+    db_port: int = Opt.db_port,
+    db_host: str = Opt.db_host,
+    db_user: str = Opt.db_user,
+) -> None:
+    """Run benchmark in the DB under test"""
+
+    info(f'Running "{scenario}" scenario')
+
+    try:
+        db = sql.Database(port=db_port, host=db_host, user=db_user)
+        db.set_database(scenario)
+
+        df = pd.DataFrame(
+            data={
+                query.name(): np.array(
+                    [
+                        db.explain(query, timing=True).optimization_time().astype(int)
+                        for _ in range(samples)
+                    ]
+                )
+                for query in [
+                    sql.Query(query)
+                    for query in sql.parse_from_file(sql.workload_path(scenario))
+                ]
+            }
+        )
+
+        if print_results:
+            print(df.to_string())
+
+        results_path = util.results_path(repository, scenario, db.mz_version())
+        info(f'Writing results to "{results_path}"')
+        df.to_csv(results_path, index=False, quoting=csv.QUOTE_MINIMAL)
+    except Exception as e:
+        err(f"run command failed: {e}")
+        raise typer.Exit()
+
+
+@app.command()
+def compare(
+    base: Path = Arg.base,
+    diff: Path = Arg.diff,
+) -> None:
+    """Compare the results of a base and diff benchmark runs."""
+
+    info(f'Compare experiment results between "{base}" and "{diff}"')
+
+    try:
+        base_df = pd.read_csv(base, quoting=csv.QUOTE_MINIMAL).agg(
+            [np.min, np.median, np.max]
+        )
+
+        diff_df = pd.read_csv(diff, quoting=csv.QUOTE_MINIMAL).agg(
+            [np.min, np.median, np.max]
+        )
+
+        # compute diff/base quotient for all (metric, query) pairs
+        quot_df = diff_df / base_df
+        # append average quotient across all queries for each metric
+        quot_df["Avg"] = quot_df.mean(axis=1)
+
+        # TODO: use styler to color-code the cells
+        print("base times")
+        print(base_df)
```

Comment:
* Should we get rid of the following output `[3 rows x 22 columns]`?
* Should we configure print to not skip columns? e.g. `...` in

```
              Q01         Q02        Q03         Q04         Q05        Q06         Q07  ...         Q16         Q17         Q18         Q19         Q20         Q21         Q22
amin    3543250.0  46156625.0  7884166.0  16175250.0  16496333.0  4742291.0  21245583.0  ...  25003291.0  32887458.0  21692458.0  20626333.0  62272541.0  49813000.0  42172833.0
```

Reply:
Done.